python/ray/services.py

```diff
@@ -267,9 +267,19 @@ def start_ray_process(command,
     if os.environ.get(tmux_env_var) == "1":
         logger.info("Detected environment variable '%s'.", tmux_env_var)
         use_tmux = True
+    gdb_env_var = "RAY_{}_GDB".format(process_type.upper())
+    if os.environ.get(gdb_env_var) == "1":
+        logger.info("Detected environment variable '%s'.", gdb_env_var)
+        use_gdb = True
 
     if use_gdb:
-        raise NotImplementedError
+        # cwd is RAY_DIRECTORY/python/ray
+        cwd = os.path.abspath(os.path.dirname(__file__))
+        gdb_file = cwd + "/.gdbinit"
```

Comment:
A couple things here:
- paths should be joined with `os.path.join`
- instead of putting it in the current working directory, can we just generate a temp file using `tempfile`?
- we should remove the file when we don't need it